tests/data/annotation_import/test_generic_data_types.py

```diff
@@ -0,0 +1,336 @@
+    parsed_t = datetime.datetime.fromisoformat(
+        date_string)  # this will blow up if the string is not in iso format
+    assert parsed_t.hour is not None
+    assert parsed_t.minute is not None
+    assert parsed_t.second is not None
+
+
+def to_pascal_case(name: str) -> str:
+    return "".join([word.capitalize() for word in name.split("_")])
+
+
+def set_project_media_type_from_data_type(project, data_type_class):
+    data_type_string = data_type_class.__name__[:-4].lower()
+    media_type = to_pascal_case(data_type_string)
+    if media_type == "Conversation":
+        media_type = "Conversational"
+    elif media_type == "Llmpromptcreation":
+        media_type = "LLMPromptCreation"
+    elif media_type == "Llmpromptresponsecreation":
+        media_type = "LLMPromptResponseCreation"
+    elif media_type == "Llmresponsecreation":
+        media_type = "Text"
+    elif media_type == "Genericdatarow":
+        media_type = "Image"
+    project.update(media_type=MediaType[media_type])
+
+
+@pytest.mark.parametrize(
+    "data_type_class",
+    [
+        AudioData,
+        HTMLData,
+        ImageData,
+        TextData,
+        VideoData,
+        ConversationData,
+        DocumentData,
+        DicomData,
+        LlmPromptCreationData,
+        LlmPromptResponseCreationData,
+        LlmResponseCreationData,
+    ],
+)
+def test_import_data_types_v2(
+    client,
+    configured_project,
+    initial_dataset,
+    data_row_json_by_data_type,
+    annotations_by_data_type_v2,
+    data_type_class,
+    exports_v2_by_data_type,
+    export_v2_test_helpers,
+    rand_gen,
+    helpers,
+):
+    project = configured_project
+    dataset = initial_dataset
+    project_id = project.uid
+
+    set_project_media_type_from_data_type(project, data_type_class)
+
+    data_type_string = data_type_class.__name__[:-4].lower()
+    data_row_ndjson = data_row_json_by_data_type[data_type_string]
+    data_row = create_data_row_for_project(project, dataset, data_row_ndjson,
+                                           rand_gen(str))
+    annotations_ndjson = annotations_by_data_type_v2[data_type_string]
+    annotations_list = [
+        label.annotations
+        for label in NDJsonConverter.deserialize(annotations_ndjson)
+    ]
+    labels = [
+        lb_types.Label(data={'uid': data_row.uid}, annotations=annotations)
+        for annotations in annotations_list
+    ]
+
+    label_import = lb.LabelImport.create_from_objects(
+        client, project_id, f"test-import-{data_type_string}", labels)
+    label_import.wait_until_done()
+
+    assert label_import.state == AnnotationImportState.FINISHED
+    assert len(label_import.errors) == 0
+
+    # TODO need to migrate project to the new BATCH mode and change this code
+    # to be similar to tests/integration/test_task_queue.py
+
+    result = export_v2_test_helpers.run_project_export_v2_task(project)
+    find_data_row = lambda dr: dr['data_row']['id'] == data_row.uid
```

Comment:
Kind of a nit, but can we avoid redefining this filter function? We have 3 copies of it between here and test_data_type. Something in helpers like:

def find_data_row_filter(data_row):
    return lambda dr: dr['data_row']['id'] == data_row.uid